src/cljc/witan/send/params.cljc

```diff
@@ -340,10 +342,15 @@
         valid-settings (s/calculate-valid-settings-for-need-ay valid-states)]
     (reduce (fn [coll [ay state]]
               (let [[need setting] (s/need-setting state)
+                    valid-trans (get valid-transitions setting)
                     obs (get-in observations [ay state])
                     ay-obs (get observations-per-ay ay)
-                    settings (get valid-settings [ay need])
-                    prior (->> (zipmap (vec settings) (repeat (/ 1.0 (count settings))))
+                    settings (->> (get valid-settings [ay need])
+                                  vec
+                                  (into valid-trans)
+                                  u/duplicates
+                                  vec)
```

Comment:
what is going on here? I'm not sure why there are 2 calls to `vec` and what `duplicates` is supposed to be doing.

Reply:
Getting the valid-settings and valid-trans into the same format then combining them, and removing the duplicates, then putting it back into the format the next part of the process expects (a vector)

Reply:
you can remove actual duplicates by doing `(into #{} <your vector>)`

Reply:
Whoops not removing duplicates, keeping them!